macros/dbt_expectations/schema_tests/_generalized/equal_expression.sql

```diff
@@ -10,8 +10,91 @@
     where
         {{ row_condition }}
     {% endif %}
+    {%- if group_by|length > 1 or group_by[0] != "'col'" %}
     group by
         {% for g in group_by -%}
-            {{ g }}{% if not loop.last %}, {% endif %}
+            {{ g }}{% if not loop.last %},{% endif %}
         {% endfor %}
-{% endmacro -%}
\ No newline at end of file
+    {% endif %}
+{% endmacro -%}
+
+{%- macro sqlserver__test_equal_expression(model, expression,
+                                compare_model,
+                                compare_expression,
+                                group_by,
+                                compare_group_by,
+                                row_condition,
+                                compare_row_condition,
+                                tolerance,
+                                tolerance_percent,
+                                return_difference) -%}
+
+    {%- set compare_model = model if not compare_model else compare_model -%}
+    {%- set compare_expression = expression if not compare_expression else compare_expression -%}
+    {%- set compare_row_condition = row_condition if not compare_row_condition else compare_row_condition -%}
+    {%- set compare_group_by = group_by if not compare_group_by else compare_group_by -%}
+
+    {%- set n_cols = group_by|length %}
```

Comment:
@b-per maybe this is the needed change?
```suggestion
    {%- set n_cols = (group_by|length) if group_by else 0 %}
```